hyperspy/signal.py

```diff
@@ -4081,18 +4110,18 @@ def diff(self, axis, order=1):
         >>> s.diff(-1).data.shape
         (64,64,1023)
         """
-
-        s = self._deepcopy_with_new_data(
-            np.diff(self.data,
-                    n=order,
-                    axis=self.axes_manager[axis].index_in_array))
-        axis = s.axes_manager[axis]
-        axis.offset += (order * axis.scale / 2)
+        s = out or self._deepcopy_with_new_data(None)
+        s.data = np.diff(self.data, n=order,
+                         axis=self.axes_manager[axis].index_in_array)
```

Comment:
Here I would also do as previously:
```python
if out:
    out.data[:] = data
else:
    s.data = data
```
To get the "passive numpy shape checking"...